main.tf

```
@@ -56,5 +56,16 @@ resource "google_cloudfunctions_function" "function" {
     }
   }
 
+  dynamic "secret_environment_variables" {
+    for_each = var.secret_environment_variables != null ? [true] : []
```

just a suggestion: `var.secret_environment_variables` can default to `[]` to make it easier..

```suggestion
    for_each = var.secret_environment_variables
```

---

I was following the existing style here. I am adjusting my code of course, but would you also mean to change other existing cases like that too (event_trigger, source_repository)?

---

it was just a suggestion... we can keep it as is to not change out of scope... will resolve the comment...